### PR TITLE
fix(shorebird_cli): look after -- separator for target and flavor

### LINKED ISSUE #1654: fix: `shorebird release android` fails if flavor argument is passed after `--` separator
**Description**

I'm getting this error when trying to create the first release.

I also have notice that there is a new folder `productionRelease` and inside it there is a file `app-production-release.aab`

```
$ shorebird release android -- --release --flavor=production -t lib/main_prod.dart


✓ Building release (95.3s)
✓ Fetching apps (0.4s)
✗ Exception: Failed to extract version name from app bundle: [BT:1.15.6] Error: File '/Users/eb/Documents/repos/vitat/health-platform-app/build/app/outputs/bundle/release/app-release.aab' was not found.
java.lang.IllegalArgumentException: File '/Users/eb/Documents/repos/vitat/health-platform-app/build/app/outputs/bundle/release/app-release.aab' was not found.
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:220)
        at com.android.tools.build.bundletool.model.utils.files.FilePreconditions.checkFileExistsAndReadable(FilePreconditions.java:35)
        at com.android.tools.build.bundletool.commands.DumpCommand.validateInput(DumpCommand.java:167)
        at com.android.tools.build.bundletool.commands.DumpCommand.execute(DumpCommand.java:142)
        at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:129)
        at com.android.tools.build.bundletool.BundleToolMain.main(BundleToolMain.java:55)
```

<details>
<summary>shorebird doctor -v</summary>
Shorebird 0.22.0 • git@github.com:shorebirdtech/shorebird.git
Flutter 3.16.7 • revision 0a7a66cbc9850f7bd7fe9c36a2f9b9084fa73a13
Engine • revision a070b8d394991f4f4457fedf2246038120847841

Android Toolchain
  • Android Studio: /Applications/Android Studio.app/Contents
  • Android SDK: /Users/eb/Library/Android/sdk
  • ADB: /Users/eb/Library/Android/sdk/platform-tools/adb
  • JAVA_HOME: /Applications/Android Studio.app/Contents/jbr/Contents/Home

✓ Shorebird is up-to-date (0.5s)
✓ Flutter install is correct (0.7s)
✓ AndroidManifest.xml files contain INTERNET permission (34ms)
✓ Has access to storage.googleapis.com (0.1s)

No issues detected!
</details>

<details>
<summary>flutter doctor -v</summary>
[✓] Flutter (Channel stable, 3.16.7, on macOS 13.0 22A8380 darwin-arm64, locale en-BR)
    • Flutter version 3.16.7 on channel stable at /Users/eb/.puro/envs/v3_16_7/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ef1af02aea (8 days ago), 2024-01-11 15:19:26 -0600
    • Engine revision 4a585b7929
    • Dart version 3.2.4
    • DevTools version 2.28.5

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/eb/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME = /Users/eb/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] VS Code (version 1.85.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.81.20231227

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.0 22A8380 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 120.0.6099.234

[✓] Network resources
    • All expected network resources are available.

• No issues found!
</details>

**Steps To Reproduce**

run: `✗ shorebird release android -- --release --flavor=production -t lib/main_prod.dart --dart-define verbose=true`

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Updates patch and release commands to look for `target` and `flavor` parameters passed as part of `rest` (i.e., after the `--` arguments separator).

Fixes https://github.com/shorebirdtech/shorebird/issues/1654

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart [hunk 1]
```diff
@@ -105,8 +106,8 @@ If this option is not provided, the version number will be determined from the p
     await cache.updateAll();
 
     const platform = ReleasePlatform.android;
-    final flavor = results['flavor'] as String?;
-    final target = results['target'] as String?;
+    final flavor = results.findOption('flavor', argParser: argParser);
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart:109]:
```suggestion
    // Detect flags even when passed to underlying commands via a `--` separator.
    // e.g. `shorebird release ios -- --flavor=foo` we still want to know about `flavor`.
    final flavor = results.findOption('flavor', argParser: argParser);
```

File: packages/shorebird_cli/lib/src/commands/patch/patch_ios_command.dart [hunk 2]
```diff
@@ -162,14 +163,15 @@ If this option is not provided, the version number will be determined from the p
...
+    final flavor = results.findOption('flavor', argParser: argParser);
+    final target = results.findOption('target', argParser: argParser);
 
     final shorebirdYaml = shorebirdEnv.getShorebirdYaml()!;
     final appId = shorebirdYaml.getAppId(flavor: flavor);
     final app = await codePushClientWrapper.getApp(appId: appId);
 
     try {
-      await _buildPatch();
+      await _buildPatch(flavor: flavor, target: target);
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/patch/patch_ios_command.dart:174]: Why are these now passed to buildPatch and they were not before?

**Review by eseidel — APPROVED** [review]

File: packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/patch/patch_android_command.dart:109]: I think this should probably be a doc comment on findOption, which I for some reason did not add. Updating now.

File: packages/shorebird_cli/lib/src/commands/patch/patch_ios_command.dart (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/patch/patch_ios_command.dart:174]: This makes it so we're only looking up the arg once. Minor optimization, but seemed better 🤷

**Pull request merged**